pkg/workflows/ring/plugin_test.go

```diff
@@ -0,0 +1,353 @@
+package ring
```

Comment:
I'll read the tests once the other comments are addressed, since the tests will likely change.